### PR TITLE
Update Batch SDK to use new API version 2018-12-01

### PR DESCRIPTION
**Pull request by xingwu1**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by xingwu1** [comment]
@matthchr @bgklein  Please review the change.

File: azure-batch/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,32 @@
 Release History
 ===============
 
+6.0.0 (2018-12-14)
+++++++++++++++++++
+
+- Using REST API version 2018-12-01.8.0.
+  - **Breaking** Removed support for the `upgrade_os` API on `CloudServiceConfiguration` pools.
```
Comment (matthchr) [inline azure-batch/HISTORY.rst]: Use `[` around breaking (match with below)

**Comment by bgklein** [comment]
Did we want to ping/have @lmazuel run his magical script to collapse models to one file. Should make doc's a little cleaner and solve the import issue Jake ran into.

File: azure-batch/azure/batch/operations/task_operations.py [hunk 2]
```diff
@@ -253,8 +255,8 @@ def add_collection(
         tasks which failed to add, a client can retry the request. In a retry,
         it is most efficient to resubmit only tasks that failed to add, and to
         omit tasks that were successfully added on the first attempt. The
-        maximum lifetime of a task from addition to completion is 7 days. If a
-        task has not completed within 7 days of being added it will be
+        maximum lifetime of a task from addition to completion is 180 days. If
```
[the remark this answers is not in the document]
Comment (xingwu1) [reply azure-batch/azure/batch/operations/task_operations.py]: Fixed.

**Comment by lmazuel** [comment]
@bgklein I think you should indeed :)

File: azure-batch/azure/__init__.py [hunk 3]
```diff
@@ -1 +1 @@
-__path__ = __import__('pkgutil').extend_path(__path__, __name__)
\ No newline at end of file
+__import__('pkg_resources').declare_namespace(__name__)
```
Comment (lmazuel) [inline azure-batch/azure/__init__.py]: Not ok, keep the pkgutil file

File: azure-batch/azure/batch/batch_service_client.py [hunk 4]
```diff
@@ -34,23 +34,26 @@ class BatchServiceClientConfiguration(AzureConfiguration):
...
     :type credentials: :mod:`A msrestazure Credentials
      object<msrestazure.azure_active_directory>`
-    :param str base_url: Service URL
+    :param batch_url: The base URL for all Azure Batch service requests.
+    :type batch_url: str
     """
 
     def __init__(
-            self, credentials, base_url=None):
+            self, credentials, batch_url):
```
[the remark this answers is not in the document]
Comment (xingwu1) [reply azure-batch/azure/batch/batch_service_client.py]: Batch endpoint is <user account>.<region>.batch.azure.com. So there is never a default endpoint. So new code is correct after I fix this issue in swagger spec.

File: azure-batch/azure/__init__.py (hunk 3 above)
Comment (xingwu1) [reply azure-batch/azure/__init__.py]: Fiexed.

**Comment by xingwu1** [comment]
@lmazuel Do you have the script? How can I run it?

**Comment by xingwu1** [comment]
@lmazuel Let me know if there has anything to change.

File: azure-batch/azure/__init__.py (hunk 3 above)
Comment (lmazuel) [reply azure-batch/azure/__init__.py]: @xingwu1 You need to remove line 1, right now your file has two lines, one with pkgresources, one with pkgutil

Comment (xingwu1) [reply azure-batch/azure/__init__.py]: Fixed.

**Comment by bgklein** [comment]
Changes look good to me.  @matthchr is OOF until the new year. We should be good to merge and release

**Comment by lmazuel** [comment]
@bgklein "should" ? :). May I humbly ask you a more assertive sentence about this, since once it's released, it cannot be undone :)

**Comment by xingwu1** [comment]
@lmazuel please merge it.

**Pull request merged**